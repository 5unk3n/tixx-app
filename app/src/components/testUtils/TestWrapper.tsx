import { NavigationContainer } from '@react-navigation/native'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import React, { PropsWithChildren } from 'react'
import { PaperProvider } from 'react-native-paper'

import { theme } from '@/theme/defaultTheme'

const queryClient = new QueryClient({
	defaultOptions: {
		queries: {
			retry: false,
			staleTime: Infinity
		}
	}
})

export const TestWrapper = ({ children }: PropsWithChildren) => {
	return (
		<PaperProvider theme={theme.dark}>
			<QueryClientProvider client={queryClient}>
				<NavigationContainer>{children}</NavigationContainer>
			</QueryClientProvider>
		</PaperProvider>
	)
}
