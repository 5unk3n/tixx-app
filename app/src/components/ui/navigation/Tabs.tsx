import React, { createContext, useContext } from 'react'
import { View } from 'react-native'
import { TouchableRipple } from 'react-native-paper'

import { CustomText } from '../display/CustomText'

interface TabsContextType<T extends string> {
	value: T
	onChange: (value: T) => void
	type?: 'underline' | 'outline' | 'dot'
}

const TabsContext = createContext<TabsContextType<any> | null>(null)

function useTabsContext<T extends string>() {
	const context = useContext(TabsContext)
	if (!context) {
		throw new Error(
			'Tab compound components must be used within Tabs component'
		)
	}
	return context as TabsContextType<T>
}

interface TabsProps<T extends string> extends TabsContextType<T> {
	children: React.ReactNode
}

export default function Tabs<T extends string>({
	value,
	onChange,
	type = 'underline',
	children,
	...props
}: TabsProps<T>) {
	const classNames = {
		tabsUnderline: 'flex-row sticky bg-background',
		tabsOutline: 'flex-row gap-3',
		tabsDot: 'flex-row gap-2'
	}

	return (
		<TabsContext.Provider value={{ value, onChange, type }}>
			<View
				className={
					type === 'underline'
						? classNames.tabsUnderline
						: type === 'outline'
							? classNames.tabsOutline
							: classNames.tabsDot
				}
				{...props}
			>
				{children}
			</View>
		</TabsContext.Provider>
	)
}

// TODO: 타입 추론 되게 수정하기
interface TabProps<T extends string> {
	value: T
	label: string
}

function Tab<T extends string>({ value, label, ...props }: TabProps<T>) {
	const { value: currentValue, onChange, type } = useTabsContext<T>()
	const isActive = value === currentValue

	const classNames = {
		underline: 'flex-1 py-3 items-center border-b-[1px] border-grayscale-700',
		underlineActive:
			'flex-1 py-3 items-center border-b-[1px] border-grayscale-700 border-grayscale-100',
		underlineText: 'text-grayscale-600',
		underlineTextActive: 'text-grayscale-100',
		outline: 'px-[19] py-[9] rounded-[20px] overflow-hidden',
		outlineActive:
			'px-[18] py-[8] border border-primary rounded-[20px] overflow-hidden',
		outlineText: 'text-grayscale-0',
		outlineTextActive: 'text-grayscale-0',
		dot: 'px-[19] py-[9]',
		dotActive: 'px-[19] py-[9]',
		dotText: 'text-grayscale-500',
		dotTextActive: 'text-grayscale-0'
	}

	return (
		<TouchableRipple
			key={value}
			onPress={() => onChange(value)}
			className={
				type === 'underline'
					? isActive
						? classNames.underlineActive
						: classNames.underline
					: type === 'outline'
						? isActive
							? classNames.outlineActive
							: classNames.outline
						: isActive
							? classNames.dotActive
							: classNames.dot
			}
			borderless
			{...props}
		>
			<View>
				<CustomText
					variant="body1Medium"
					className={
						type === 'underline'
							? isActive
								? classNames.underlineTextActive
								: classNames.underlineText
							: type === 'outline'
								? isActive
									? classNames.outlineTextActive
									: classNames.outlineText
								: isActive
									? classNames.dotTextActive
									: classNames.dotText
					}
				>
					{label}
				</CustomText>
				{type === 'dot' && isActive && (
					<CustomText className="text-center">•</CustomText>
				)}
			</View>
		</TouchableRipple>
	)
}

Tabs.Tab = Tab
