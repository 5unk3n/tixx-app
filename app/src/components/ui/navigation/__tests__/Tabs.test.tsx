import { describe, it, expect, jest } from '@jest/globals'
import { render, screen, fireEvent } from '@testing-library/react-native'
import React from 'react'

import { TestWrapper } from '../../../testUtils/TestWrapper'
import Tabs from '../Tabs'

describe('Tabs', () => {
	it('should render tabs and handle selection change', () => {
		const onChangeMock = jest.fn()
		render(
			<Tabs value="a" onChange={onChangeMock}>
				<Tabs.Tab value="a" label="Tab A" />
				<Tabs.Tab value="b" label="Tab B" />
			</Tabs>,
			{ wrapper: TestWrapper }
		)

		expect(screen.getByText('Tab A')).toBeTruthy()
		expect(screen.getByText('Tab B')).toBeTruthy()

		fireEvent.press(screen.getByText('Tab B'))
		expect(onChangeMock).toHaveBeenCalledWith('b')
	})

	describe('should render correctly based on type and active state', () => {
		const types = ['underline', 'outline', 'dot'] as const
		const values = ['a', 'b']

		types.forEach((type) => {
			values.forEach((value) => {
				it(`renders correctly for type '${type}' with active tab '${value}'`, () => {
					const { toJSON } = render(
						<Tabs value={value} onChange={jest.fn()} type={type}>
							<Tabs.Tab value="a" label="Tab A" />
							<Tabs.Tab value="b" label="Tab B" />
						</Tabs>,
						{ wrapper: TestWrapper }
					)
					expect(toJSON()).toMatchSnapshot()
				})
			})
		})
	})
})
