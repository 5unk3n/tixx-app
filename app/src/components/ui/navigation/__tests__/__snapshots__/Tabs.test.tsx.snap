// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly based on type and active state renders correctly for type 'dot' with active tab 'a' 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "marginLeft": -8,
            "marginTop": -8,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 19,
                "paddingRight": 19,
              },
              {
                "paddingBottom": 9,
                "paddingTop": 9,
              },
              [
                {
                  "marginLeft": 8,
                  "marginTop": 8,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#FFFFFF",
                    },
                  ],
                ],
              ]
            }
          >
            Tab A
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Pretendard-Medium",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": -0.02,
                  "lineHeight": 18,
                },
                {
                  "writingDirection": "ltr",
                },
                [
                  {
                    "textAlign": "center",
                  },
                ],
              ]
            }
          >
            â€¢
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 19,
                "paddingRight": 19,
              },
              {
                "paddingBottom": 9,
                "paddingTop": 9,
              },
              [
                {
                  "marginLeft": 8,
                  "marginTop": 8,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#7E7D7D",
                    },
                  ],
                ],
              ]
            }
          >
            Tab B
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Tabs should render correctly based on type and active state renders correctly for type 'dot' with active tab 'b' 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "marginLeft": -8,
            "marginTop": -8,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 19,
                "paddingRight": 19,
              },
              {
                "paddingBottom": 9,
                "paddingTop": 9,
              },
              [
                {
                  "marginLeft": 8,
                  "marginTop": 8,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#7E7D7D",
                    },
                  ],
                ],
              ]
            }
          >
            Tab A
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 19,
                "paddingRight": 19,
              },
              {
                "paddingBottom": 9,
                "paddingTop": 9,
              },
              [
                {
                  "marginLeft": 8,
                  "marginTop": 8,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#FFFFFF",
                    },
                  ],
                ],
              ]
            }
          >
            Tab B
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Pretendard-Medium",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": -0.02,
                  "lineHeight": 18,
                },
                {
                  "writingDirection": "ltr",
                },
                [
                  {
                    "textAlign": "center",
                  },
                ],
              ]
            }
          >
            â€¢
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Tabs should render correctly based on type and active state renders correctly for type 'outline' with active tab 'a' 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "marginLeft": -12,
            "marginTop": -12,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 18,
                "paddingRight": 18,
              },
              {
                "paddingBottom": 8,
                "paddingTop": 8,
              },
              {
                "borderBottomWidth": 1,
                "borderLeftWidth": 1,
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              {
                "borderBottomColor": "#F2F862",
                "borderLeftColor": "#F2F862",
                "borderRightColor": "#F2F862",
                "borderTopColor": "#F2F862",
              },
              {
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
              },
              {
                "overflow": "hidden",
              },
              [
                {
                  "marginLeft": 12,
                  "marginTop": 12,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#FFFFFF",
                    },
                  ],
                ],
              ]
            }
          >
            Tab A
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 19,
                "paddingRight": 19,
              },
              {
                "paddingBottom": 9,
                "paddingTop": 9,
              },
              {
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
              },
              {
                "overflow": "hidden",
              },
              [
                {
                  "marginLeft": 12,
                  "marginTop": 12,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#FFFFFF",
                    },
                  ],
                ],
              ]
            }
          >
            Tab B
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Tabs should render correctly based on type and active state renders correctly for type 'outline' with active tab 'b' 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "marginLeft": -12,
            "marginTop": -12,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 19,
                "paddingRight": 19,
              },
              {
                "paddingBottom": 9,
                "paddingTop": 9,
              },
              {
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
              },
              {
                "overflow": "hidden",
              },
              [
                {
                  "marginLeft": 12,
                  "marginTop": 12,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#FFFFFF",
                    },
                  ],
                ],
              ]
            }
          >
            Tab A
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "paddingLeft": 18,
                "paddingRight": 18,
              },
              {
                "paddingBottom": 8,
                "paddingTop": 8,
              },
              {
                "borderBottomWidth": 1,
                "borderLeftWidth": 1,
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              {
                "borderBottomColor": "#F2F862",
                "borderLeftColor": "#F2F862",
                "borderRightColor": "#F2F862",
                "borderTopColor": "#F2F862",
              },
              {
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
              },
              {
                "overflow": "hidden",
              },
              [
                {
                  "marginLeft": 12,
                  "marginTop": 12,
                },
              ],
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#FFFFFF",
                    },
                  ],
                ],
              ]
            }
          >
            Tab B
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Tabs should render correctly based on type and active state renders correctly for type 'underline' with active tab 'a' 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "backgroundColor": "#000000",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              {
                "alignItems": "center",
              },
              {
                "borderBottomWidth": 1,
              },
              {
                "borderBottomColor": "#2F2F30",
                "borderLeftColor": "#2F2F30",
                "borderRightColor": "#2F2F30",
                "borderTopColor": "#2F2F30",
              },
              {
                "borderBottomColor": "#EDEDED",
                "borderLeftColor": "#EDEDED",
                "borderRightColor": "#EDEDED",
                "borderTopColor": "#EDEDED",
              },
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#EDEDED",
                    },
                  ],
                ],
              ]
            }
          >
            Tab A
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              {
                "alignItems": "center",
              },
              {
                "borderBottomWidth": 1,
              },
              {
                "borderBottomColor": "#2F2F30",
                "borderLeftColor": "#2F2F30",
                "borderRightColor": "#2F2F30",
                "borderTopColor": "#2F2F30",
              },
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#585858",
                    },
                  ],
                ],
              ]
            }
          >
            Tab B
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Tabs should render correctly based on type and active state renders correctly for type 'underline' with active tab 'b' 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "backgroundColor": "#000000",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              {
                "alignItems": "center",
              },
              {
                "borderBottomWidth": 1,
              },
              {
                "borderBottomColor": "#2F2F30",
                "borderLeftColor": "#2F2F30",
                "borderRightColor": "#2F2F30",
                "borderTopColor": "#2F2F30",
              },
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#585858",
                    },
                  ],
                ],
              ]
            }
          >
            Tab A
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              {
                "alignItems": "center",
              },
              {
                "borderBottomWidth": 1,
              },
              {
                "borderBottomColor": "#2F2F30",
                "borderLeftColor": "#2F2F30",
                "borderRightColor": "#2F2F30",
                "borderTopColor": "#2F2F30",
              },
              {
                "borderBottomColor": "#EDEDED",
                "borderLeftColor": "#EDEDED",
                "borderRightColor": "#EDEDED",
                "borderTopColor": "#EDEDED",
              },
            ],
          ]
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#FFFFFF",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "Pretendard-Medium",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "letterSpacing": -0.02,
                    "lineHeight": 18,
                  },
                  [
                    {
                      "color": "#EDEDED",
                    },
                  ],
                ],
              ]
            }
          >
            Tab B
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
